src/main/java/pitt/search/semanticvectors/lsh/InMemoryLSHStore.java

```diff
@@ -55,15 +58,18 @@ public void initCache() throws IOException {
 		}
 
 		VectorStoreReaderLucene vecStore = new VectorStoreReaderLucene(vecStoreFile.getAbsolutePath(), flagConfig);
-		short[] potentialHashes = lsh.getSimilarHashes(vector);
-		Set<Long> allVecPositions = new HashSet<>();
-		for (short potentialHash : potentialHashes) {
-			if (storeHash.containsKey(potentialHash))
-				allVecPositions.addAll(storeHash.get(potentialHash));
+		try {
+			short[] potentialHashes = lsh.getSimilarHashes(vector);
+			Set<Long> allVecPositions = new HashSet<>();
+			for (short potentialHash : potentialHashes) {
+				if (storeHash.containsKey(potentialHash))
+					allVecPositions.addAll(storeHash.get(potentialHash));
+			}
+			Iterator<Long> iter = allVecPositions.iterator();
+			return getObjectVectorEnum(iter, vecStore, flagConfig);
+		} finally {
+			vecStore.close();
```

Comment:
Why don't use VectorStoreUtils.closeVectorStores(vecStore); here instead of vectorStore.close()